validator/client/validator.go

```diff
@@ -1030,36 +1034,50 @@ func (v *validator) PushProposerSettings(ctx context.Context, km keymanager.IKey
 
 	return nil
 }
-
-func (v *validator) buildPrepProposerReqs(ctx context.Context, pubkeys [][fieldparams.BLSPubkeyLength]byte) ([]*ethpb.PrepareBeaconProposerRequest_FeeRecipientContainer, error) {
-	var prepareProposerReqs []*ethpb.PrepareBeaconProposerRequest_FeeRecipientContainer
-
+func (v *validator) filterAndCacheActiveKeys(ctx context.Context, pubkeys [][fieldparams.BLSPubkeyLength]byte) ([][fieldparams.BLSPubkeyLength]byte, error) {
+	filteredKeys := make([][fieldparams.BLSPubkeyLength]byte, 0)
 	for _, k := range pubkeys {
-		// Default case: Define fee recipient to burn address
-		var feeRecipient common.Address
-		isFeeRecipientDefined := false
-
-		// If fee recipient is defined in default configuration, use it
-		if v.ProposerSettings() != nil && v.ProposerSettings().DefaultConfig != nil && v.ProposerSettings().DefaultConfig.FeeRecipientConfig != nil {
-			feeRecipient = v.ProposerSettings().DefaultConfig.FeeRecipientConfig.FeeRecipient // Use cli config for fee recipient.
-			isFeeRecipientDefined = true
-		}
-
-		validatorIndex, ok := v.pubkeyToValidatorIndex[k]
+		_, ok := v.pubkeyToValidatorIndex[k]
 		// Get validator index from RPC server if not found.
 		if !ok {
 			i, ok, err := v.validatorIndex(ctx, k)
 			if err != nil {
 				return nil, err
 			}
-
 			if !ok { // Nothing we can do if RPC server doesn't have validator index.
 				continue
 			}
-
-			validatorIndex = i
 			v.pubkeyToValidatorIndex[k] = i
 		}
+		status, err := v.validatorClient.ValidatorStatus(ctx, &ethpb.ValidatorStatusRequest{PublicKey: k[:]})
```

Comment:
doing this for every validator seems expensive, each request will be a separate roundtrip to the beacon node.

Reply:
I've updated using our internal API which will minimize the requests.